File: tests/Generator/integration/php/Location.php
<?php
use PSX\Schema\Attribute\Description;
use PSX\Schema\Attribute\Required;

#[Description('Location of the person')]
#[Required(array('lat', 'long'))]
class Location implements \JsonSerializable, \PSX\Record\RecordableInterface
{
    protected ?float $lat = null;
    protected ?float $long = null;
    public function setLat(?float $lat) : void
    {
        $this->lat = $lat;
    }
    public function getLat() : ?float
    {
        return $this->lat;
    }
    public function setLong(?float $long) : void
    {
        $this->long = $long;
    }
    public function getLong() : ?float
    {
        return $this->long;
    }
    public function toRecord() : \PSX\Record\RecordInterface
    {
        /** @var \PSX\Record\Record<mixed> $record */
        $record = new \PSX\Record\Record();
        $record->put('lat', $this->lat);
        $record->put('long', $this->long);
        return $record;
    }
    public function jsonSerialize() : object
    {
        return (object) $this->toRecord()->getAll();
    }
}

File: tests/Generator/integration/java/Location.java
package org.phpsx.test;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonSetter;

/**
 * Location of the person
 */
public class Location {
    private double lat;
    private double _long;
    @JsonSetter("lat")
    public void setLat(double lat) {
        this.lat = lat;
    }
    @JsonGetter("lat")
    public double getLat() {
        return this.lat;
    }
    @JsonSetter("long")
    public void setLong(double _long) {
        this._long = _long;
    }
    @JsonGetter("long")
    public double getLong() {
        return this._long;
    }
}
